xchange-stream-service-netty/src/main/java/info/bitrich/xchangestream/service/netty/NettyStreamingService.java

```diff
@@ -266,18 +267,22 @@ protected void initChannel(SocketChannel ch) {
             })
         .doOnComplete(
             () -> {
-              LOG.warn("Resubscribing channels");
               resubscribeChannels();
 
               connectionSuccessEmitters.forEach(emitter -> emitter.onNext(new Object()));
             });
   }
 
   private void scheduleReconnect() {
-    LOG.info("Scheduling reconnection");
-    webSocketChannel
-        .eventLoop()
-        .schedule(() -> connect().subscribe(), retryDuration.toMillis(), TimeUnit.MILLISECONDS);
+    if (autoReconnect) {
+      LOG.info("Scheduling reconnection");
+      webSocketChannel.eventLoop().schedule(
+              () -> connect().subscribe(),
+              retryDuration.toMillis(),
+              TimeUnit.MILLISECONDS);
+    } else {
+      reconnFailEmitters.forEach(emitter -> emitter.onNext(new Exception("AutoReconnect disabled")));
```

Comment:
Hm, this is an interesting choice. I'm not sure about the idea of reporting a "reconnection failure" when no reconnection was attempted, or desired. Could you explain this choice?